### PR TITLE
feat(cloudsql): Add postgres samples for enterprise & enterprise_plus edition

### PR DESCRIPTION
**Pull request by ibhaskar2**
## Description
Added documentation for cloud sql instance edition

Fixes #<ISSUE-NUMBER>

Note: If you are not associated with Google, open an issue for discussion before submitting a pull request.

## Checklist

**Readiness**

- [X] Yes, **merge** this PR after it is approved
- [ ] No, don't **merge** this PR after it is approved

**Style**

- [X] My sample follows the rules described for Terraform in the [Effective Samples style guide](https://googlecloudplatform.github.io/samples-style-guide/)
- [X] My sample follows the other requirements and best practices in the [Contributing
guide](https://github.com/terraform-google-modules/terraform-docs-samples/blob/main/CONTRIBUTING.md#other-requirements-and-best-practices)

**Testing**

- [X] I have performed tests described in the [Contributing guide](https://github.com/terraform-google-modules/terraform-docs-samples/blob/main/CONTRIBUTING.md):

   - [X] **[Tests](https://github.com/terraform-google-modules/terraform-docs-samples/blob/main/CONTRIBUTING.md#set-up-the-test-environment)** pass: `terraform apply`
   - [X] **[Lint](https://github.com/terraform-google-modules/terraform-docs-samples/blob/main/CONTRIBUTING.md#linting-and-formatting)** pass: `terraform fmt` check

**Intended location**

- [X] Yes, this sample will be (or already is) included on cloud.google.com
      Location(s):https://cloud.google.com/sql/docs/editions-intro

- [ ] No, this sample won't be included on cloud.google.com
      Reason:

**API enablement**

- [ ] If the sample needs an API enabled to pass testing, I have added the service to the [Test setup file](https://github.com/terraform-google-modules/terraform-docs-samples/blob/main/test/setup/main.tf)

**Review**

- [ ] If this sample adds a new directory, I have added codeowners to the [CODEOWNERS file](https://github.com/terraform-google-modules/terraform-docs-samples/blob/main/.github/CODEOWNERS)

### CODE REVIEW

**Comment by gericdong** [comment]
@ibhaskar2 please fix the lint and test errors when you have a chance. Thanks.

File: cloud_sql/instance_edition [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+resource "google_sql_database_instance" "enterprise_instance" {
+  name             = "enterprise-postgres-instance"
+  region           = "us-central1"
+  database_version = "POSTGRES_15"
+  settings {
+    tier = "db-g1-small"
+  }
+  deletion_protection = "false"
+}
+# [END cloud_sql_enterprise_instance]
```
Comment (enocom) [inline cloud_sql/instance_edition:40]: nit missing newline

**Review by enocom — APPROVED** [review]

File: cloud_sql/instance_edition (hunk 1 above)
Comment (ibhaskar2) [reply cloud_sql/instance_edition]: Fixed it

**Review by rogerthatdev — CHANGES_REQUESTED** [review]
please add the sample to `main.tf` file in the `cloud_sql/instance_edition/` directory.

**Comment by ibhaskar2** [comment]
@rogerthatdev please review it, i have made required changes

**Comment by msampathkumar** [comment]
@ibhaskar2 Almost there! Use `mysql_` and `postgres_` prefixes in the folder names.

* cloud_sql/enterprise_edition_instance/main.tf --> cloud_sql/mysql_enterprise_edition_instance/main.tf

* cloud_sql/enterprise_plus_edition_instance/main.tf --> cloud_sql/postgres_enterprise_plus_edition_instance/main.tf

File: cloud_sql/enterprise_edition_instance/main.tf [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+# [START cloud_sql_enterprise_instance]
+resource "google_sql_database_instance" "main" {
```
Comment (msampathkumar) [inline cloud_sql/enterprise_edition_instance/main.tf:19]:
```suggestion
resource "google_sql_database_instance" "default" {
```

File: cloud_sql/enterprise_plus_edition_instance/main.tf [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+# [START cloud_sql_enterprise_plus_instance]
+resource "google_sql_database_instance" "main" {
```
Comment (msampathkumar) [inline cloud_sql/enterprise_plus_edition_instance/main.tf:18]:
```suggestion
resource "google_sql_database_instance" "default" {
```

File: cloud_sql/enterprise_edition_instance/main.tf (hunk 2 above)
Comment (ibhaskar2) [reply cloud_sql/enterprise_edition_instance/main.tf]: Done

File: cloud_sql/enterprise_plus_edition_instance/main.tf (hunk 3 above)
Comment (ibhaskar2) [reply cloud_sql/enterprise_plus_edition_instance/main.tf]: Done

**Review by msampathkumar — APPROVED** [review]
LGTM.

/gcbrun

**Review by rogerthatdev — APPROVED** [review]

**Pull request merged**